R/transactions.R

```diff
@@ -43,4 +57,29 @@ setGeneric("dbCommit",
 setGeneric("dbRollback",
   def = function(conn, ...) standardGeneric("dbRollback"),
   valueClass = "logical"
-)
\ No newline at end of file
+)
+
+#' @aliases withTransaction,DBIConnection-method
+#' @export
+#' @rdname transactions
+setGeneric("withTransaction",
+  def = function(conn, statement, ...) standardGeneric("withTransaction"),
+  valueClass = "logical"
+)
+
+#' @export
+setMethod("withTransaction", "DBIConnection", function(conn, statement, ...) {
```

Comment:
Using a function would also be a more natural parallel with `dbFetchChunked()` as that callback needs to be passed the data frame. (Unless you really want to go gung-ho with NSE and use some sort of pronoun in the expression, which definitely doesn't feel like DBI)